executor/index_lookup_hash_join.go

```diff
@@ -174,6 +201,51 @@ func (e *IndexNestedLoopHashJoin) Next(ctx context.Context, req *chunk.Chunk) er
 	return nil
 }
 
+func (e *IndexNestedLoopHashJoin) runInOrder(ctx context.Context, req *chunk.Chunk) error {
+	var (
+		result *indexHashJoinResult
+		ok     bool
+	)
+	for {
+		task := e.fetchNextTask(ctx)
+		if task == nil {
+			return nil
+		}
+		select {
+		case result, ok = <-task.resultCh:
+			if !ok {
```

Comment:
Should we return here since this channel is closed?

Reply:
Noop, when the task.resultCh is closed, it only means that the task is finished and we need to fetch the next task.